lib/modules/datasource/npm/get.ts

```diff
@@ -218,29 +237,32 @@ export async function getDependency(
     }
     return dep;
   } catch (err) {
+    const actualError = err instanceof ExternalHostError ? err.err : err;
     const ignoredStatusCodes = [401, 402, 403, 404];
     const ignoredResponseCodes = ['ENOTFOUND'];
     if (
-      err.message === HOST_DISABLED ||
-      ignoredStatusCodes.includes(err.statusCode) ||
-      ignoredResponseCodes.includes(err.code)
+      actualError.message === HOST_DISABLED ||
+      ignoredStatusCodes.includes(actualError.statusCode) ||
+      ignoredResponseCodes.includes(actualError.code)
     ) {
       return null;
     }
-    if (uri.host === 'registry.npmjs.org') {
+
+    if (err instanceof ExternalHostError) {
       if (cachedResult) {
         logger.warn(
-          { err },
-          'npmjs error, reusing expired cached result instead',
+          { err, host: uri.host },
+          `npm host error, reusing expired cached result instead`,
         );
         delete cachedResult.cacheData;
         return cachedResult;
       }
-      // istanbul ignore if
-      if (err.name === 'ParseError' && err.body) {
-        err.body = 'err.body deleted by Renovate';
+
+      if (actualError instanceof ParseError) {
+        const wrappedError = actualError as ParseError & { body?: string };
+        wrappedError.body = wrappedError.body && 'err.body deleted by Renovate';
```

Comment:
I took this over from the existing code: 
```
      if (err.name === 'ParseError' && err.body) {
        err.body = 'err.body deleted by Renovate';
 ```
 
 This will only add `'err.body deleted by Renovate'` if a body existed. If the error had no body, it won't be added.

Reply:
this code isn't the same